In [1]:
import pandas as pd

In [2]:
counties_ts_df = pd.read_csv("../Data/counties.timeseries.csv")
# counties_ts_df.describe()

In [4]:
counties_ts_df.columns.tolist()

['date',
 'country',
 'state',
 'county',
 'fips',
 'lat',
 'long',
 'locationId',
 'actuals.cases',
 'actuals.deaths',
 'actuals.positiveTests',
 'actuals.negativeTests',
 'actuals.contactTracers',
 'actuals.hospitalBeds.capacity',
 'actuals.hospitalBeds.currentUsageTotal',
 'actuals.hospitalBeds.currentUsageCovid',
 'unused1',
 'actuals.icuBeds.capacity',
 'actuals.icuBeds.currentUsageTotal',
 'actuals.icuBeds.currentUsageCovid',
 'unused2',
 'actuals.newCases',
 'actuals.vaccinesDistributed',
 'actuals.vaccinationsInitiated',
 'actuals.vaccinationsCompleted',
 'metrics.testPositivityRatio',
 'metrics.testPositivityRatioDetails',
 'metrics.caseDensity',
 'metrics.contactTracerCapacityRatio',
 'metrics.infectionRate',
 'metrics.infectionRateCI90',
 'unused3',
 'unused4',
 'metrics.icuCapacityRatio',
 'riskLevels.overall',
 'metrics.vaccinationsInitiatedRatio',
 'metrics.vaccinationsCompletedRatio',
 'actuals.newDeaths',
 'actuals.vaccinesAdministered',
 'riskLevels.caseDensity',
 'cdc

In [7]:
counties_ts_df.tail()

date state         county   fips  actuals.cases  \
2547371  2022-05-10    WY  Weston County  56045         1588.0   
2547372  2022-05-11    WY  Weston County  56045         1588.0   
2547373  2022-05-12    WY  Weston County  56045         1588.0   
2547374  2022-05-13    WY  Weston County  56045         1588.0   
2547375  2022-05-14    WY  Weston County  56045            NaN   

         actuals.deaths  actuals.positiveTests  actuals.negativeTests  \
2547371            18.0                    NaN                    NaN   
2547372            18.0                    NaN                    NaN   
2547373            18.0                    NaN                    NaN   
2547374            18.0                    NaN                    NaN   
2547375             NaN                    NaN                    NaN   

         actuals.contactTracers  actuals.hospitalBeds.capacity  ...  \
2547371                     NaN                            NaN  ...   
2547372                     NaN                            NaN  ...   
2547373                     NaN                            NaN  ...   
2547374                     NaN                            NaN  ...   
2547375                     NaN                            NaN  ...   

         actuals.hsaIcuBeds.capacity  actuals.hsaIcuBeds.currentUsageTotal  \
2547371                          NaN                                   NaN   
2547372                          NaN                                   NaN   
2547373                          NaN                                   NaN   
2547374                          NaN                                   NaN   
2547375                          NaN                                   NaN   

         actuals.hsaIcuBeds.currentUsageCovid  \
2547371                                   NaN   
2547372                                   NaN   
2547373                                   NaN   
2547374                                   NaN   
2547375                                   NaN   

         metrics.bedsWithCovidPatientsRatio     hsaName  \
2547371                                 NaN  Weston, WY   
2547372                                 NaN  Weston, WY   
2547373                                 0.0  Weston, WY   
2547374                                 NaN  Weston, WY   
2547375                                 NaN  Weston, WY   

         actuals.hospitalBeds.weeklyCovidAdmissions  \
2547371                                         NaN   
2547372                                         NaN   
2547373                                         NaN   
2547374                                         NaN   
2547375                                         NaN   

         actuals.hsaHospitalBeds.weeklyCovidAdmissions  \
2547371                                            NaN   
2547372                                            NaN   
2547373                                            NaN   
2547374                                            NaN   
2547375                                            NaN   

         metrics.weeklyCovidAdmissionsPer100k  \
2547371                                   NaN   
2547372                                   NaN   
2547373                                   0.0   
2547374                                   NaN   
2547375                                   NaN   

         communityLevels.cdcCommunityLevel  communityLevels.canCommunityLevel  
2547371                                NaN                                0.0  
2547372                                NaN                                0.0  
2547373                                0.0                                0.0  
2547374                                NaN                                0.0  
2547375                                NaN                                0.0  

[5 rows x 50 columns]

In [5]:
counties_ts_df.dropna(how="all",axis=1,inplace=True)
print(counties_ts_df.shape)
counties_ts_df.drop(columns=['country','locationId'],inplace=True)
counties_ts_df.info()

(2547376, 52)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2547376 entries, 0 to 2547375
Data columns (total 50 columns):
 #   Column                                         Dtype  
---  ------                                         -----  
 0   date                                           object 
 1   state                                          object 
 2   county                                         object 
 3   fips                                           int64  
 4   actuals.cases                                  float64
 5   actuals.deaths                                 float64
 6   actuals.positiveTests                          float64
 7   actuals.negativeTests                          float64
 8   actuals.contactTracers                         float64
 9   actuals.hospitalBeds.capacity                  float64
 10  actuals.hospitalBeds.currentUsageTotal         float64
 11  actuals.hospitalBeds.currentUsageCovid         float64
 12  actuals.icuBeds.capacity    

In [9]:
final_df = counties_ts_df[["date",'fips',"state","county","actuals.cases","actuals.positiveTests","actuals.newCases","actuals.deaths","actuals.newDeaths","metrics.caseDensity","actuals.hospitalBeds.currentUsageTotal",
                        "actuals.hospitalBeds.currentUsageCovid","metrics.weeklyCovidAdmissionsPer100k","actuals.icuBeds.currentUsageTotal",
                        "actuals.icuBeds.currentUsageCovid","metrics.vaccinationsCompletedRatio", "metrics.vaccinationsInitiatedRatio", "metrics.infectionRate", 
                        "riskLevels.overall","riskLevels.caseDensity", "cdcTransmissionLevel", "metrics.testPositivityRatio"]]

In [11]:
final_df.describe()

fips  actuals.cases  actuals.positiveTests  actuals.newCases  \
count  2.547376e+06   2.480835e+06             794.000000      2.457057e+06   
mean   3.134935e+04   1.002057e+04           74772.366499      3.272756e+01   
std    1.632679e+04   4.394017e+04           66035.254700      2.965221e+02   
min    1.001000e+03   0.000000e+00               1.000000      0.000000e+00   
25%    1.901900e+04   3.860000e+02           20298.250000      0.000000e+00   
50%    3.001900e+04   1.783000e+03           64644.500000      2.000000e+00   
75%    4.610300e+04   5.910000e+03           91996.250000      1.300000e+01   
max    7.215300e+04   2.908425e+06          225776.000000      1.104410e+05   

       actuals.deaths  actuals.newDeaths  metrics.caseDensity  \
count    2.425775e+06       2.415311e+06         2.496369e+06   
mean     1.615211e+02       4.005716e-01         3.126351e+01   
std      6.713339e+02       2.703575e+00         4.704566e+01   
min      0.000000e+00       0.000000e+00         0.000000e+00   
25%      6.000000e+00       0.000000e+00         4.500000e+00   
50%      3.300000e+01       0.000000e+00         1.470000e+01   
75%      1.020000e+02       0.000000e+00         3.980000e+01   
max      3.202200e+04       6.740000e+02         3.477400e+03   

       actuals.hospitalBeds.currentUsageTotal  \
count                           252158.000000   
mean                               215.139567   
std                                678.233814   
min                                  0.000000   
25%                                  8.000000   
50%                                 22.000000   
75%                                124.000000   
max                              16186.000000   

       actuals.hospitalBeds.currentUsageCovid  \
count                           178091.000000   
mean                                32.326631   
std                                113.195462   
min                                  0.000000   
25%                                  2.000000   
50%                                  6.000000   
75%                                 24.000000   
max                               7887.000000   

       metrics.weeklyCovidAdmissionsPer100k  \
count                         239457.000000   
mean                              18.887772   
std                               22.482526   
min                                0.000000   
25%                                5.500000   
50%                               12.300000   
75%                               25.200000   
max                             1048.300000   

       actuals.icuBeds.currentUsageTotal  actuals.icuBeds.currentUsageCovid  \
count                      148168.000000                      115124.000000   
mean                           42.982594                          11.895035   
std                           102.500180                          31.749218   
min                             0.000000                           0.000000   
25%                             4.000000                           1.000000   
50%                            10.000000                           4.000000   
75%                            34.000000                          11.000000   
max                          2640.000000                        1697.000000   

       metrics.vaccinationsCompletedRatio  metrics.vaccinationsInitiatedRatio  \
count                        1.416273e+06                        1.404100e+06   
mean                         3.758096e-01                        4.383015e-01   
std                          1.872815e-01                        1.956141e-01   
min                          0.000000e+00                        0.000000e+00   
25%                          2.550000e-01                        3.130000e-01   
50%                          3.960000e-01                        4.520000e-01   
75%                          5.030000e-01                        5.660000e-01   
max                          9.500000e-01        

In [12]:
for col in final_df.columns:
    print(col, final_df[col].isna().sum())

date 0
fips 0
state 0
county 0
actuals.cases 66541
actuals.positiveTests 2546582
actuals.newCases 90319
actuals.deaths 121601
actuals.newDeaths 132065
metrics.caseDensity 51007
actuals.hospitalBeds.currentUsageTotal 2295218
actuals.hospitalBeds.currentUsageCovid 2369285
metrics.weeklyCovidAdmissionsPer100k 2307919
actuals.icuBeds.currentUsageTotal 2399208
actuals.icuBeds.currentUsageCovid 2432252
metrics.vaccinationsCompletedRatio 1131103
metrics.vaccinationsInitiatedRatio 1143276
metrics.infectionRate 176218
riskLevels.overall 0
riskLevels.caseDensity 0
cdcTransmissionLevel 0
metrics.testPositivityRatio 495898


In [13]:
final_df = final_df[["date",'fips',"state","county","actuals.cases","actuals.deaths","metrics.caseDensity","metrics.vaccinationsCompletedRatio", "metrics.vaccinationsInitiatedRatio", "metrics.infectionRate", 
                        "riskLevels.overall","riskLevels.caseDensity", "cdcTransmissionLevel", "metrics.testPositivityRatio"]]

Features Dropped:
1. Positive Tests (Cases metric instead)
2. Negative Tests 
3. New Cases and Deaths (Inconsistencies)

Too many missing Values:
1. actuals.hospitalBeds.currentUsageTotal 2295218
2. actuals.hospitalBeds.currentUsageCovid 2369285
3. metrics.weeklyCovidAdmissionsPer100k 2307919
4. actuals.icuBeds.currentUsageTotal 2399208
5. actuals.icuBeds.currentUsageCovid 2432252


In [14]:
final_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2547376 entries, 0 to 2547375
Data columns (total 14 columns):
 #   Column                              Dtype  
---  ------                              -----  
 0   date                                object 
 1   fips                                int64  
 2   state                               object 
 3   county                              object 
 4   actuals.cases                       float64
 5   actuals.deaths                      float64
 6   metrics.caseDensity                 float64
 7   metrics.vaccinationsCompletedRatio  float64
 8   metrics.vaccinationsInitiatedRatio  float64
 9   metrics.infectionRate               float64
 10  riskLevels.overall                  int64  
 11  riskLevels.caseDensity              int64  
 12  cdcTransmissionLevel                int64  
 13  metrics.testPositivityRatio         float64
dtypes: float64(7), int64(4), object(3)
memory usage: 272.1+ MB


In [15]:
final_df.to_csv("counties.finalfeatures.timeseries.csv",index=False)

In [16]:
final_df

date   fips state                  county  actuals.cases  \
0        2020-04-09   2013    AK  Aleutians East Borough            NaN   
1        2020-04-10   2013    AK  Aleutians East Borough            NaN   
2        2020-04-11   2013    AK  Aleutians East Borough            NaN   
3        2020-04-12   2013    AK  Aleutians East Borough            NaN   
4        2020-04-13   2013    AK  Aleutians East Borough            NaN   
...             ...    ...   ...                     ...            ...   
2547371  2022-05-10  56045    WY           Weston County         1588.0   
2547372  2022-05-11  56045    WY           Weston County         1588.0   
2547373  2022-05-12  56045    WY           Weston County         1588.0   
2547374  2022-05-13  56045    WY           Weston County         1588.0   
2547375  2022-05-14  56045    WY           Weston County            NaN   

         actuals.deaths  metrics.caseDensity  \
0                   NaN                  0.0   
1                   NaN                  0.0   
2                   NaN                  0.0   
3                   NaN                  0.0   
4                   NaN                  0.0   
...                 ...                  ...   
2547371            18.0                  NaN   
2547372            18.0                  NaN   
2547373            18.0                  NaN   
2547374            18.0                  NaN   
2547375             NaN                  NaN   

         metrics.vaccinationsCompletedRatio  \
0                                       NaN   
1                                       NaN   
2                                       NaN   
3                                       NaN   
4                                       NaN   
...                                     ...   
2547371                               0.380   
2547372                               0.380   
2547373                               0.380   
2547374                               0.380   
2547375                               0.381   

         metrics.vaccinationsInitiatedRatio  metrics.infectionRate  \
0                                       NaN                    NaN   
1                                       NaN                    NaN   
2                                       NaN                    NaN   
3                                       NaN                    NaN   
4                                       NaN                    NaN   
...                                     ...                    ...   
2547371                               0.412                    1.0   
2547372                               0.412                    1.0   
2547373                               0.412                    1.0   
2547374                               0.412                    1.0   
2547375                               0.413                    NaN   

         riskLevels.overall  riskLevels.caseDensity  cdcTransmissionLevel  \
0                         0                       0                     0   
1                         0                       0                     0   
2                         0                       0                     0   
3                         0                       0                     0   
4                         0                       0                     0   
...                     ...                     ...                   ...   
2547371                   1                       1                     2   
2547372                   1                       1                     2   
2547373                   1                       1                     1   
2547374                   1                       1                     2   
2547375                   1                       1                     2   

         metrics.testPositivityRatio  
0                                NaN  
1                                NaN  
2                                NaN  
3                                NaN  
4          

## Creating Final Combined Dataframe

### Combining Population Data

In [17]:
# cleaned_states_df = pd.read_csv("states.finalfeatures.timeseries.csv")
# cleaned_states_df.shape

In [18]:
# states_info_df = pd.read_csv("./Data/states.csv")
# state_pop_df = states_info_df[["state","population"]]
# state_pop_df.set_index("state",inplace=True)

# population_list = []
# for col in cleaned_states_df["state"]:
#     population = state_pop_df.loc[col]["population"]
#     population_list.append(population)
    
# cleaned_states_df["population"] = population_list
# cleaned_states_df.shape

In [19]:
# # def create_pop_based_features():

# cleaned_counties_df['vaccinationsCompletedRatio.Density'] = cleaned_counties_df["metrics.vaccinationsCompletedRatio"]/counties_pop_df.loc[county]["population"]
    
# cleaned_counties_df['vaccinationsCompletedRatio.Density'] = items_df['Actual Price'] - \
#     ((items_df['Discount(%)']/100) * items_df['Actual Price'])

In [20]:
# cleaned_counties_df = pd.read_csv("counties.finalfeatures.timeseries.csv")
# states_final_df.to_csv("states.finalfeatures.timeseries.csv",index=False)

In [21]:
# final_states_df = pd.merge(cleaned_states_df, state_pop_df, on="state")
# final_counties_df = pd.merge(cleaned_counties_df, counties_pop_df, how='inner',on="county")

In [22]:
# final_counties_df.info(verbose=True,show_counts=True)